packages/react-core/src/components/Nav/NavExpandable.tsx

```diff
@@ -91,14 +93,33 @@ export class NavExpandable extends React.Component<NavExpandableProps, NavExpand
 
   render() {
     // eslint-disable-next-line @typescript-eslint/no-unused-vars
-    const { id, title, srText, children, className, isActive, groupId, isExpanded, onExpand, ...props } = this.props;
+    const {
+      id,
+      title,
+      srText,
+      children,
+      className,
+      isActive,
+      groupId,
+      isExpanded,
+      onExpand,
+      ouiaId,
+      ouiaSafe,
+      ...props
+    } = this.props;
 
     const { expandedState } = this.state;
 
     const onClick = () => {
       this.setState(prevState => ({ expandedState: !prevState.expandedState }));
     };
 
+    const ouiaProps = useOUIAProps(
+      NavExpandable.displayName,
+      ouiaId !== undefined ? ouiaId : getDefaultOUIAId(NavExpandable.displayName),
```

Comment:
You can just write `ouiaId`. The `useOUIAProps` function handles the case of it being undefined.

Reply:
Tests fail if I do that. I copied that pattern from here https://github.com/patternfly/patternfly-react/blob/master/packages/react-core/src/components/Nav/Nav.tsx#L152

Reply:
What tests fail? The `nav` pattern you're copying is because `Nav` uses a strange `ouiaStateId` that I don't think it needs anymore.

Reply:
```
Summary of all failing tests
 FAIL  packages/react-core/src/components/Nav/__tests__/Nav.test.tsx
  ● Expandable Nav List

    Invalid hook call. Hooks can only be called inside of the body of a function component. This could happen for one of the following reasons:
    1. You might have mismatching versions of React and the renderer (such as React DOM)
    2. You might be breaking the Rules of Hooks
    3. You might have more than one copy of React in the same app
    See https://fb.me/react-invalid-hook-call for tips about how to debug and fix this problem.

      66 |     return id;
      67 |   }
    > 68 |   return useMemo(() => getDefaultOUIAId(componentType, variant), [componentType, variant]);
         |          ^
      69 | };
      70 | 
      71 | /**

      at Object.throwInvalidHookError (node_modules/react-dom/cjs/react-dom.development.js:2527:528)
      at useMemo (node_modules/react/cjs/react.development.js:1492:25)
      at useOUIAId (packages/react-core/src/helpers/ouia.ts:68:10)
      at useOUIAProps (packages/react-core/src/helpers/ouia.ts:54:29)
      at NavExpandable.render (packages/react-core/src/components/Nav/NavExpandable.tsx:117:23)
      at finishClassComponent (node_modules/react-dom/cjs/react-dom.development.js:2768:100)
      at updateClassComponent (node_modules/react-dom/cjs/react-dom.development.js:2760:220)
      at beginWork (node_modules/react-dom/cjs/react-dom.development.js:3049:934)
      at HTMLUnknownElement.callCallback (node_modules/react-dom/cjs/react-dom.development.js:70:102)
      at Object.invokeGuardedCallbackDev (node_modules/react-dom/cjs/react-dom.development.js:90:45)
      at invokeGuardedCallback (node_modules/react-dom/cjs/react-dom.development.js:105:126)
      at beginWork$1 (node_modules/react-dom/cjs/react-dom.development.js:3934:1)
      at performUnitOfWork (node_modules/react-dom/cjs/react-dom.development.js:3734:301)
      at workLoopSync (node_modules/react-dom/cjs/react-dom.development.js:3730:45)
      at performSyncWorkOnRoot (node_modules/react-dom/cjs/react-dom.development.js:3665:223)
      at scheduleUpdateOnFiber (node_modules/react-dom/cjs/react-dom.development.js:3512:1)
      at updateContainer (node_modules/react-dom/cjs/react-dom.development.js:4087:313)
      at node_modules/react-dom/cjs/react-dom.development.js:4117:29
      at unbatchedUpdates (node_modules/react-dom/cjs/react-dom.development.js:3687:189)
      at legacyRenderSubtreeIntoContainer (node_modules/react-dom/cjs/react-dom.development.js:4117:1)
      at Object.render (node_modules/react-dom/cjs/react-dom.development.js:4119:519)
      at unmount (node_modules/enzyme-adapter-react-16/src/ReactSixteenAdapter.js:508:5)
      at node_modules/enzyme-adapter-react-16/src/ReactSixteenAdapter.js:438:7
      at batchedUpdates$1 (node_modules/react-dom/cjs/react-dom.development.js:3683:146)
      at Object.act (node_modules/react-dom/cjs/react-dom-test-utils.development.js:919:18)
      at wrapAct (node_modules/enzyme-adapter-react-16/src/ReactSixteenAdapter.js:413:12)
      at Object.render (node_modules/enzyme-adapter-react-16/src/ReactSixteenAdapter.js:499:4)
      at new ReactWrapper (node_modules/enzyme/build/ReactWrapper.js:208:16)
      at mount (node_modules/enzyme/build/mount.js:26:10)
      at Object.<anonymous> (packages/react-core/src/components/Nav/__tests__/Nav.test.tsx:98:16)

  ● Expandable Nav List - Trigger toggle

    Invalid hook call. Hooks can only be called inside of the body of a function component. This could happen for one of the following reasons:
    1. You might have mismatching versions of React and the renderer (such as React DOM)
    2. You might be breaking the Rules of Hooks
    3. You might have more than one copy of React in the same app
    See https://fb.me/react-invalid-hook-call for tips about how to debug and fix this problem.

      66 |     return id;
      67 |   }
    > 68 |   return useMemo(() => getDefaultOUIAId(componentType, variant), [componentType, variant]);
         |          ^
      69 | };
      70 | 
      71 | /**

      at Object.throwInvalidHookError (node_modules/react-dom/cjs/react-dom.development.js:2527:528)
      at useMemo (node_modules/react/cjs/react.development.js:1492:25)
      at useOUIAId (packages/react-core/src/helpers/ouia.ts:68:10)
      at useOUIAProps (packages/react-core/src/helpers/ouia.ts:54:29)
      at NavExpandable.render (packages/react-core/src/components/Nav/NavExpandable.tsx:117:23)
      at finishClassComponent (node_modules/react-dom/cjs/react-dom.development.js:2768:100)
      at updateClassComponent (node_modules/react-dom/cjs/react-dom.development.js:2760:220)
      at beginWork (node_modules/react-dom/cjs/react-dom.development.js:3049:934)
      at HTMLUnknownElement.callCallback (node_modules/react-dom/cjs/react-dom.development.js:70:102)
      at Object.invokeGuardedCallbackDev (node_modules/react-dom/cjs/react-dom.development.js:90:45)
      at invokeGuardedCallback (node_modules/react-dom/cjs/react-dom.development.js:105:126)
      at beginWork$1 (node_modules/react-dom/cjs/react-dom.development.js:3934:1)
      at performUnitOfWork (node_modules/react-dom/cjs/react-dom.development.js:3734:301)
      at workLoopSync (node_modules/react-dom/cjs/react-dom.development.js:3730:45)
      at performSyncWorkOnRoot (node_modules/react-dom/cjs/react-dom.development.js:3665:223)
      at scheduleUpdateOnFiber (node_modules/react-dom/cjs/react-dom.development.js:3512:1)
      at updateContainer (node_modules/react-dom/cjs/react-dom.development.js:4087:313)
      at node_modules/react-dom/cjs/react-dom.development.js:4117:29
      at unbatchedUpdates (node_modules/react-dom/cjs/react-dom.development.js:3687:189)
      at legacyRenderSubtreeIntoContainer (node_modules/react-dom/cjs/react-dom.development.js:4117:1)
      at Object.render (node_modules/react-dom/cjs/react-dom.development.js:4119:519)
      at unmount (node_modules/enzyme-adapter-react-16/src/ReactSixteenAdapter.js:508:5)
      at node_modules/enzyme-adapter-react-16/src/ReactSixteenAdapter.js:438:7
      at batchedUpdates$1 (node_modules/react-dom/cjs/react-dom.development.js:3683:146)
      at Object.act (node_modules/react-dom/cjs/react-dom-test-utils.development.js:919:18)
      at wrapAct (node_modules/enzyme-adapter-react-16/src/ReactSixteenAdapter.js:413:12)
      at Object.render (node_modules/enzyme-adapter-react-16/src/ReactSixteenAdapter.js:499:4)
      at new ReactWrapper (node_modules/enzyme/build/ReactWrapper.js:208:16)
      at mount (node_modules/enzyme/build/mount.js:26:10)
      at Object.<anonymous> (packages/react-core/src/components/Nav/__tests__/Nav.test.tsx:117:16)

  ● Expandable Nav List with aria label

    Invalid hook call. Hooks can only be called inside of the body of a function component. This could happen for one of the following reasons:
    1. You might have mismatching versions of React and the renderer (such as React DOM)
    2. You might be breaking the Rules of Hooks
    3. You might have more than one copy of React in the same app
    See https://fb.me/react-invalid-hook-call for tips about how to debug and fix this problem.

      66 |     return id;
      67 |   }
    > 68 |   return useMemo(() => getDefaultOUIAId(componentType, variant), [componentType, variant]);
         |          ^
      69 | };
      70 | 
      71 | /**

      at Object.throwInvalidHookError (node_modules/react-dom/cjs/react-dom.development.js:2527:528)
      at useMemo (node_modules/react/cjs/react.development.js:1492:25)
      at useOUIAId (packages/react-core/src/helpers/ouia.ts:68:10)
      at useOUIAProps (packages/react-core/src/helpers/ouia.ts:54:29)
      at NavExpandable.render (packages/react-core/src/components/Nav/NavExpandable.tsx:117:23)
      at finishClassComponent (node_modules/react-dom/cjs/react-dom.development.js:2768:100)
      at updateClassComponent (node_modules/react-dom/cjs/react-dom.development.js:2760:220)
      at beginWork (node_modules/react-dom/cjs/react-dom.development.js:3049:934)
      at HTMLUnknownElement.callCallback (node_modules/react-dom/cjs/react-dom.development.js:70:102)
      at Object.invokeGuardedCallbackDev (node_modules/react-dom/cjs/react-dom.development.js:90:45)
      at invokeGuardedCallback (node_modules/react-dom/cjs/react-dom.development.js:105:126)
      at beginWork$1 (node_modules/react-dom/cjs/react-dom.development.js:3934:1)
      at performUnitOfWork (node_modules/react-dom/cjs/react-dom.development.js:3734:301)
      at workLoopSync (node_modules/react-dom/cjs/react-dom.development.js:3730:45)
      at performSyncWorkOnRoot (node_modules/react-dom/cjs/react-dom.development.js:3665:223)
      at scheduleUpdateOnFiber (node_modules/react-dom/cjs/react-dom.development.js:3512:1)
      at updateContainer (node_modules/react-dom/cjs/react-dom.development.js:4087:313)
      at node_modules/react-dom/cjs/react-dom.development.js:4117:29
      at unbatchedUpdates (node_modules/react-dom/cjs/react-dom.development.js:3687:189)
      at legacyRenderSubtreeIntoContainer (node_modules/react-dom/cjs/react-dom.development.js:4117:1)
      at Object.render (node_modules/react-dom/cjs/react-dom.development.js:4119:519)
      at unmount (node_modules/enzyme-adapter-react-16/src/ReactSixteenAdapter.js:508:5)
      at node_modules/enzyme-adapter-react-16/src/ReactSixteenAdapter.js:438:7
      at batchedUpdates$1 (node_modules/react-dom/cjs/react-dom.development.js:3683:146)
      at Object.act (node_modules/react-dom/cjs/react-dom-test-utils.development.js:919:18)
      at wrapAct (node_modules/enzyme-adapter-react-16/src/ReactSixteenAdapter.js:413:12)
      at Object.render (node_modules/enzyme-adapter-react-16/src/ReactSixteenAdapter.js:499:4)
      at new ReactWrapper (node_modules/enzyme/build/ReactWrapper.js:208:16)
      at mount (node_modules/enzyme/build/mount.js:26:10)
      at Object.<anonymous> (packages/react-core/src/components/Nav/__tests__/Nav.test.tsx:139:16)

 FAIL  packages/react-core/src/components/Nav/__tests__/Generated/NavExpandable.test.tsx
  ● NavExpandable should match snapshot (auto-generated)

    Invalid hook call. Hooks can only be called inside of the body of a function component. This could happen for one of the following reasons:
    1. You might have mismatching versions of React and the renderer (such as React DOM)
    2. You might be breaking the Rules of Hooks
    3. You might have more than one copy of React in the same app
    See https://fb.me/react-invalid-hook-call for tips about how to debug and fix this problem.

      66 |     return id;
      67 |   }
    > 68 |   return useMemo(() => getDefaultOUIAId(componentType, variant), [componentType, variant]);
         |          ^
      69 | };
      70 | 
      71 | /**

      at resolveDispatcher (node_modules/react/cjs/react.development.js:1431:17)
      at useMemo (node_modules/react/cjs/react.development.js:1491:24)
      at useOUIAId (packages/react-core/src/helpers/ouia.ts:68:10)
      at useOUIAProps (packages/react-core/src/helpers/ouia.ts:54:29)
      at NavExpandable.render (packages/react-core/src/components/Nav/NavExpandable.tsx:117:23)
      at ReactShallowRenderer._mountClassComponent (node_modules/react-test-renderer/cjs/react-test-renderer-shallow.development.js:878:41)
      at ReactShallowRenderer.render (node_modules/react-test-renderer/cjs/react-test-renderer-shallow.development.js:795:18)
      at node_modules/enzyme-adapter-react-16/src/ReactSixteenAdapter.js:768:23
      at withSetStateAllowed (node_modules/enzyme-adapter-utils/src/Utils.js:210:15)
      at Object.render (node_modules/enzyme-adapter-react-16/src/ReactSixteenAdapter.js:769:18)
      at new ShallowWrapper (node_modules/enzyme/src/ShallowWrapper.js:475:4)
      at shallow (node_modules/enzyme/build/shallow.js:26:10)
      at Object.<anonymous> (packages/react-core/src/components/Nav/__tests__/Generated/NavExpandable.test.tsx:11:16)
```

Reply:
Ah, this is because you're trying to use a hook in a class component. You could convert this class to a function component which uses `const [expandedState, setExpandedState] = React.useState();` for its state, or sit tight while @jschuler and I find a pattern for class components.

Reply:
Let's wait for @jschuler

Reply:
https://github.com/patternfly/patternfly-react/blob/303722fb41ee9327abfc05b9088a7ff246b327bc/packages/react-core/src/helpers/ouia.ts#L21
for class based components:
```
 * In class based components, create a state variable ouiaStateId to create a static generated ID:
 * state = {
 *  ouiaStateId: getDefaultOUIAId(Chip.displayName)
 * }
 * This generated ID should remain alive as long as the component is not unmounted.
 *
 * Then add the attributes to the component
 * {...getOUIAProps('OverflowChip', this.props.ouiaId !== undefined ? this.props.ouiaId : this.state.ouiaStateId)}
```